ext/node/ops/tls.rs

```diff
@@ -68,3 +102,492 @@ pub fn op_tls_canonicalize_ipv4_address(
 
   Some(canonical_ip)
 }
+
+struct ReadableFuture<'a> {
+  socket: &'a JSStreamSocket,
+}
+
+impl<'a> Future for ReadableFuture<'a> {
+  type Output = std::io::Result<()>;
+
+  fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+    self.socket.poll_read_ready(cx)
+  }
+}
+
+struct WritableFuture<'a> {
+  socket: &'a JSStreamSocket,
+}
+
+impl<'a> Future for WritableFuture<'a> {
+  type Output = std::io::Result<()>;
+
+  fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+    self.socket.poll_write_ready(cx)
+  }
+}
+
+#[derive(Debug)]
+pub struct JSStreamSocket {
```

Comment:
Nit why `JSStreamSocket` instead of `JsStreamSocket`?

Reply:
It's called `JSStreamSocket` in Node.js